File: framework/languages/node/e2e/ToActorMessaging/Client/Scenarios/ta-b1-missing-actor-scenario.ts
// TA-B1: 존재하지 않는 Actor를 호출한다 시나리오를 검증한다.
import type { ClientOptions } from '../Support/client-options';
import { getJson } from '../../../http-client';
import {
  type ActorEvidence, assertFailure, requireNoEvidence
} from '../Support/actor-scenario-support';

export async function runTaB1(options: ClientOptions): Promise<void> {
  await assertFailure(options, 'TA-B1-missing-send', 'ta-b1-missing', 'NotFound', true);
  await assertFailure(options, 'TA-B1-missing-request', 'ta-b1-missing', 'NotFound', false);
  requireNoEvidence(await getJson<ActorEvidence[]>(`${options.actorUrl}/evidence`), 'TA-B1-missing-send');
  console.log('scenario TA-B1 passed');
}

File: framework/languages/dotnet/e2e/ToActorMessaging/Client/Scenarios/TaB1MissingActorScenario.cs
// Verifies TA-B1 Missing Actor behavior.
using ToActorMessaging.Client.Support;

namespace ToActorMessaging.Client.Scenarios;

internal static class TaB1MissingActorScenario
{
    public static async Task RunAsync(ToActorScenarioContext context)
    {
        const string actorId = "missing-actor";
        await context.AssertRouteAbsentAsync(actorId);
        //  The send's submit result is not the verification here (config 9
        //  TA-B1). What must hold is that no handler ran and no authority was
        //  created, which the evidence and route checks below assert.
        await context.SendWithoutOutcomeAssertionAsync(
            "TA-B1-missing",
            actorId,
            "missing",
            targetNodeRid: "actor-a",
            targetGeneration: 1);
        await context.AssertFailureAsync(
            "TA-B1-missing-request",
            actorId,
            "NotFound",
            send: false,
            targetNodeRid: "actor-a",
            targetGeneration: 1);
        await context.AssertNoActorEvidenceAsync(actorId);
        await context.AssertRouteAbsentAsync(actorId);
    }
}
